source/common/http/http2/codec_impl.h

```diff
@@ -520,12 +520,13 @@ class ConnectionImpl : public virtual Connection,
     void encodeTrailers(const ResponseTrailerMap& trailers) override {
       encodeTrailersBase(trailers);
     }
+    void setRequestDecoder(Http::RequestDecoder& decoder) override { request_decoder_ = &decoder; }
```

Comment:
Is request_decoder_ assigned somewhere already? Please make it private and use this setter in those call sites.

Reply:
Made this private. The one assignment was already changed to using this setter.